### DISCUSSION TITLE
How to make tables in Markor & Markdown?

### DISCUSSION THREAD

**Discussion by ClaudioEnriqueRossi** [gsantner/markor#1715] [Q&A] [answered]
When I try to generate a table in Markdown mode, regardless of the number of columns, from the six box icon on the bottom icon line of the screen, it does not build the entire table, ie the header and below the horizontal division line.

Just it do this, for example I chose 3 columns:

|  | 

And this considering hold down to generate the header etc. Thanks.

**Comment by gsantner** [comment, marked as answer]
How to make a table:

1) Long press/hold the table button to make the table header with amount of  wanted columns
2) Place the cursor where you want to add a new row and select the same amount of columns (when you have multiple rows already you can insert somewhere in the middle. when you want to insert at the end, place cursor at last row)
3) (short) press the table for as much rows you need

![screenrecord-2022-05-22_19 38 33](https://user-images.githubusercontent.com/6735650/169708288-ee14cf27-e193-4912-9db0-0f9b378044e2.gif)

**Reply by ClaudioEnriqueRossi** [reply]
No, bad answer, I was clear in my statement about what is the problem with creating tables.

**Reply by gsantner** [reply]
```
Just it do this, for example I chose 3 columns:

| |
```

this is the correct insert of  a table row. Not a table header. Not sure what other issue you have here, but as you can see in the video it works just fine in the current Markor version.

**Reply by ClaudioEnriqueRossi** [reply]
No, it's not the answer, and don't answer with that because it's clear what I'm saying.

**Reply by gsantner** [reply]
As you see it is not. I'm not here to waste time guessing. Maybe be more concrete, with pictures, what your issues is. I don't see an issue here with tables.

**Reply by soovitamstele** [reply]
Didn't work for me either but persevere it's worth it. It's just as fast to do it manually. 

For the header write your col names separated by vertical bar (asc 124) then carriage return CR  
Type 3 minus signs (asc 45) followed by vertical bar then CR  
Now enter row data, columns separated by vertical bar  
This is the minimum to create a table I think, as below. 

Too late for the OP but might be useful for someone struggling like me. You can copy and paste the example below into a markor note and press the eye button to see it rendered. Then play with different vertical bar positions. It's very flexible. 

Important note: markor renders more "generously" than this webbot does.  For example cell merging doesn't work here and line 2 needs to be filled out  ---|---|--- to match the number of columns desired for the table to render. So be aware if exporting!  HTH 

Also Markor will only render markdown when clicking the eye if the document format is set to markdown. It is the default format for new markdown files (.md)  The doc format is changed from a document window in edit mode:  top right 3 dot menu icon > File Settings > Format > Markdown

Great App. Many thanks

<pre>
   <code>
col1|col2|col3
---|
Row1|

-  -  - 
| Syntax | Description |
| ----------- | ----------- |
| Header | Title |
| Paragraph | Text |
https://www.markdownguide.org/cheat-sheet/  
https://commonmark.org/help/   

||Smoke Points ||  Boiling Points ||
Oil  | °F | °C | °F | °C
---|
Avocado oil  |570°F	 |271°C
Butter  |350°F	 |177°C
Sesame oil (unrefined)  |350°F	 |177°C |420°F |216°C
Soybean oil (refined)  |460°F	 |238°C |495°F |257°C

   </code>
  </pre>

**Comment by harshad1** [comment]
> No, bad answer

We're volunteers here. 

If you are having trouble with this it would be helpful to everyone if you could spend some time

1. Figuring this out
2. Writing a guide
3. Designing an improved workflow - doesn't even need to be code, just a sketch of how things should work

**Reply by soovitamstele** [reply]
Thanks for your comment  

Do you feel it really warrants a howto? There are plenty of authorative markdown resources out there. I'm not competent to talk about syntax nor the tool itself. I suspect my brief reply would have been enough for the OP and the scarcity of questions via Mr G imply that most people don't have problems. I included the table snippets just as a sandbox really with some urls thrown in for background.  
 
It's evident that the "not always present" table button (.md dox only?) does not work as reliably as suggested, which is a distraction. Hence my suggestion to enter the table framework manually which is no hardship and arguably faster. It's not HTML after all and  the header is the heart of the table design.   

It doesn't really need more than that to get things working. Experimentation and practice with the target markdown dialect is all that is needed to make progress after that isn't it?   

BTW I updated my post to include setting the doc format so the eye button works as expected.  
ps Grepping into files! How cool is that with regex too.
